### ISSUE TITLE
Technique PDF13

### ISSUE THREAD

**Issue by kpaciello** [w3c/wcag#3056]
I have a question regarding Technique PDF13. The description for this technique says: “The objective of this technique is to provide replacement link text via the /Alt entry in the property list for a tag. This is usually not necessary, but in some situations, additional information beyond the visible link text is needed, particularly for screen reader users. Screen readers can read visible link text, but replacing the screen text with meaningful alternate text for links in a PDF document can make links more accessible.”
 
The second sentence says, “This is usually not necessary…”. What did the author mean by this? In what situations would it not be necessary?

@awkawk I was told you may be able to assist with this one! Any insight you can provide would be greatly appreciated!

**Comment by awkawk** [comment]
@kpaciello If you have a link in an HTML web page that requires some additional context (e.g., "read more" links in a list of books or articles where there is a chance that the user won't be able to properly associate the link with the correct book or article title even though visually it is apparent) you might use aria-labelledby or aria-label to add additional information to the link.

That's what this is in PDF. Usually you don't need to add additional information to links in PDF, but if you do then the /Alt entry provides a mechanism to do so.

The situations where it isn't necessary are 99+% of the time. If the link text is "product documentation" or "https://www.w3.org" usually that link is just fine and what the user should hear.

**Comment by kpaciello** [comment]
@awkawk Thank you for the quick reply! I've always added Alt text for links and been taught that it's a standard remediation practice, but when I saw Technique PDF13, it made me think otherwise and your reply seems to confirm that thought. Is there a reason that PAC 2021 and CommonLook Validator both flag missing Alt text for links as failures according to WCAG 1.1.1?

ALT: PAC 2021 checker reporting an error of "Alternative description missing for an annotation" according to WCAG guideline 1.1.1
![pac-wcag-error](https://user-images.githubusercontent.com/126114980/220798334-b8788256-e2cc-4e59-b206-641f02219174.png)

ALT: CommonLook Validator reporting errors of "Parent tag of link annotation doesn't define the Alt attribute" and "Link Annotation doesn't define the Contents attribute" according to WCAG guideline 1.1
![CL-wcag-error](https://user-images.githubusercontent.com/126114980/220798331-95086bac-f112-4cec-ad72-db7b9bdf87a0.png)

**Comment by awkawk** [comment]
They include it because it is in PDF-UA version one. This was a mistake in my opinion in that it not only creates unneeded extra work, but in many cases encourages the addition of information that is unhelpful or in some cases harmful because it disallows the user to read the text presented visually.

This is expected to be fixed in PDF/UA-2.

**Comment by kpaciello** [comment]
@awkawk Thanks so much for this explanation! This topic has sparked a great conversation internally and your insight has been crucial! Out of curiosity, did you author this technique?

**Comment by awkawk** [comment]
@kpaciello I don't recall whether this was one that I authored or if it was one of the several that we had contract help to author. Looking at the images with Boston Globe references it could have been me or Jeanne Spellman possibly? Either way, it was reviewed by the Working Group.

**Comment by kpaciello** [comment]
@awkawk Hi Andrew, one more question on this topic! I reached out to Commonlook to find out why their validator flags links with no alt text. 

They said: “Technically speaking, the functionality of the link – the link Annotation, or the OBJR – is non-text content.  It’s true, in the vast majority of links, there is also text right there with the annotation but the annotation itself is the non-text content.  That’s why Alt text is required.”

So they're saying that this Link – OBJR tag (see screenshot below) is what qualifies as non-text content and therefore requires ALT text. 

ALT: Link - OBJR tag nested inside of a Link tag.
![link](https://user-images.githubusercontent.com/126114980/221589463-cf849754-600e-4000-b3dc-646ce7541717.png)

Any thoughts on this?

**Comment by awkawk** [comment]
@kpaciello The PDF spec indicates:

> For annotation types that normally display text, that text (specified in the Contents entry of the annotation dictionary) is the natural source for vocalization purposes. For annotation types that do not display text, a Contents entry (PDF 1.4) can optionally be included to specify an alternate description. 


It also includes:

> As an example, consider the following fragment of HTML code, which produces a line of text containing a hypertext link:
> ```
> <html>
>   <body>
>     <p>Here is some text <a href=http://www.adobe.com>with a link</a> inside.
>   </body>
> </html>
> ```
> Example 10.17 shows an equivalent fragment of PDF using a link element, whose text it displays in blue and underlined. Example 10.18 shows an excerpt from the associated logical structure hierarchy.

So, the PDF spec compares the link structure to HTML links, so perhaps we should be considering HTML anchor elements to be non-text content that requires an alt attribute also? Of course not. In fact, 1.1.1 doesn't require an _alt attribute_ it requires a _text alternative_ and there is no reason that we wouldn't consider the text provided along with the link annotation to serve as that alternative. It is that text that is provided to the accessibility API to be conveyed as the accessible name of the link, so an additional /alt entry is only needed when the text of the link provided needs to be augmented (don't forget that in order to pass 2.5.3 Label in name the visible text needs to be part of the accessible name) and requiring it on all links is a waste of people's effort.

**Comment by awkawk** [comment]
@kpaciello ok to close?

**Comment by kpaciello** [comment]
@awkawk Yes it is ok to close! Thanks again for your help!
